python/sparkdl/transformers/tf_tensor.py

```diff
@@ -0,0 +1,140 @@
+
+    Restrictions of the current API:
+
+    We assume that
+    - All graphs have a "minibatch" dimension (i.e. an unknown leading
+      dimension) in the tensor shapes.
+    - Input DataFrame has an array column where all elements have the same length
+    """
+
+    @keyword_only
+    def __init__(self, tfCheckpointDir=None, tfGraph=None,
+                 exportDir=None, tagSet=None, signatureDefKey=None,
+                 inputMapping=None, outputMapping=None, tfHParms=None):
+        """
+        __init__(self, tfCheckpointDir=None, tfGraph=None,
+                 exportDir=None, tagSet=None, signatureDefKey=None,
+                 inputMapping=None, outputMapping=None, tfHParms=None)
+        """
+        super(TFTransformer, self).__init__()
+        kwargs = self._input_kwargs
+        self.setParams(**kwargs)
+
+
+    @keyword_only
+    def setParams(self, tfCheckpointDir=None, tfGraph=None,
+                  exportDir=None, tagSet=None, signatureDefKey=None,
+                  inputMapping=None, outputMapping=None, tfHParms=None):
+        """
+        setParams(self, tfCheckpointDir=None, tfGraph=None,
+                  exportDir=None, tagSet=None, signatureDefKey=None,
+                  inputMapping=None, outputMapping=None, tfHParms=None)
+        """
+        super(TFTransformer, self).__init__()
+        kwargs = self._input_kwargs
+        return self._set(**kwargs)
+
+
+    def _convertInternal(self):
+        assert self.isDefined(self.inputMapping) and self.isDefined(self.outputMapping), \
+            "inputMapping and outputMapping must be defined"
+
+        _maybe_graph = self.getTFGraph()
+        with IsolatedSession(graph=_maybe_graph) as issn:
+            if self.isDefined(self.exportDir):
+                assert _maybe_graph is None
+                assert not self.isDefined(self.tfCheckpointDir)
+                tag_set = self.getTagSet().split(',')
+                tf.saved_model.loader.load(issn.sess, tag_set, self.getExportDir())
+            elif self.isDefined(self.tfCheckpointDir):
+                assert _maybe_graph is None
+                ckpt_dir = self.getTFCheckpointDir()
+                ckpt_path = tf.train.latest_checkpoint(ckpt_dir)
+                print('using checkpoint path from {} as {}'.format(ckpt_dir, ckpt_path))
+                saver = tf.train.import_meta_graph("{}.meta".format(ckpt_path), clear_devices=True)
+                saver.restore(issn.sess, ckpt_path)
+            else:
+                assert _maybe_graph is not None
+
+            sig_def = None
+            if self.isDefined(self.signatureDefKey):
+                meta_graph_def = tf.train.export_meta_graph(issn.graph)
+                sig_def = tf.contrib.saved_model.get_signature_def_by_key(
+                    meta_graph_def, self.getSignatureDefKey())
+                assert sig_def is not None
+
+            feeds = []
+            for _, tnsr_or_sig in self.getInputMapping():
+                if sig_def:
```

Comment:
When a `signature_def` is provided, the semantics of `inputMapping` and `outputMapping` change. 
Instead of mapping DataFrame columns to/from tensor/operation names, they now map columns to keys in the `signature_def` lookup table, which returns the actual tensor/operation names.

Reply:
We should probably make this process more transparent to users.